### PR TITLE
Add convenience typedefs for Date/Date32/DateTime/DateTime64 columns

### PR DESCRIPTION
**Pull request by rschu1ze**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by rschu1ze** [comment]
Let's not merge until #42308 is in (--> merge conflict).

**Review by devcrafter — APPROVED** [review]
LGTM

**Comment by rschu1ze** [comment]
Ah, I should rebase this ...

**Pull request merged**
